### PR TITLE
collect policy event for the grafana datashboard

### PR DESCRIPTION
**Pull request by yanmxa**
Signed-off-by: myan <myan@redhat.com>
Refs: https://issues.redhat.com/browse/ACM-4099
TODO: consider the batch update for a better performance.

The collected data is like this:
```bash
$ table=event.local_policies                                                                                                                                                                   
$ kubectl exec -it $(kubectl get pods -n hoh-postgres -l postgres-operator.crunchydata.com/role=master -o jsonpath='{.items..metadata.name}') -c database -n hoh-postgres -- psql -U postgres -d hoh -c "select * from $table"
              policy_id               |              cluster_id              |                                               message                                               |      reason      |
     source                |     created_at      |  compliance
--------------------------------------+--------------------------------------+-----------------------------------------------------------------------------------------------------+------------------+----------
---------------------------+---------------------+---------------
 1d858095-4689-4f47-869b-3d4761f9825b | b5fbe3fb-3059-4d28-badb-dbd9c7991e74 | Policy local-placement.policy-limitrange status was updated in cluster namespace kind-hub1-cluster1 | PolicyStatusSync | {"compone
nt": "policy-status-sync"} | 2023-05-23 12:06:24 | non_compliant
 1d858095-4689-4f47-869b-3d4761f9825b | b5fbe3fb-3059-4d28-badb-dbd9c7991e74 | Policy local-placement.policy-limitrange status was updated in cluster namespace kind-hub1-cluster1 | PolicyStatusSync | {"compone
nt": "policy-status-sync"} | 2023-05-23 12:06:17 | non_compliant
(2 rows)

```

### CODE REVIEW

**Review by morvencao — COMMENTED** [review]
Any special reason to select new kafka go lib https://github.com/Shopify/sarama?
We use github.com/confluentinc/confluent-kafka-go for spec and status syncs.

**Comment by yanmxa** [comment]
> Any special reason to select new kafka go lib https://github.com/Shopify/sarama? We use github.com/confluentinc/confluent-kafka-go for spec and status syncs.

Good question! The following are the reasons that I choose to use https://github.com/Shopify/sarama
- The implementation for cloudevents kafka protocal is https://github.com/Shopify/sarama. Considering that we may fully use cloudevents as the global hub transport in the future, we choose to be consistent with its implementation here. 
- The Shopify/sarama has a larger user base and community support, with a history of being widely adopted.

It does not say that we have to choose sarama. Confluent-kafka-go is also a good choice, but based on the current situation, the two will coexist for a while

File: manager/pkg/eventcollector/processor/policy_processor.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+func NewPolicyProcessor(ctx context.Context, pool *pgxpool.Pool, offsetManager OffsetManager) *policyProcessor {
+	return &policyProcessor{
+		log:           ctrl.Log.WithName("policy-event-processor"),
+		ctx:           ctx,
+		pool:          pool,
+		offsetManager: offsetManager,
+	}
+}
+
+func (p *policyProcessor) Process(event *kube.EnhancedEvent, eventOffset *EventOffset) {
```
Comment (clyang82) [inline manager/pkg/eventcollector/processor/policy_processor.go:35]: It is fine for the initial version to make it works. But we need another PR to handle events in a batch mode.

**Review by clyang82 — APPROVED** [review]
LGTM

**Pull request merged**